dask/array/creation.py

```diff
@@ -582,6 +582,120 @@ def tril(m, k=0):
     return Array(dsk, name, shape=m.shape, chunks=m.chunks, dtype=m.dtype)
 
 
+def np_pad(array, pad_width, mode, extra_arg=None):
+    if mode in ["maximum", "mean", "median", "minimum"]:
+        extra_arg = extra_arg or None
+        return np.pad(array, pad_width, mode, stat_length=extra_arg)
+    elif mode == "constant":
+        extra_arg = extra_arg or 0
+        return np.pad(array, pad_width, mode, constant_values=extra_arg)
+    elif mode == "linear_ramp":
+        extra_arg = extra_arg or 0
+        return np.pad(array, pad_width, mode, end_values=extra_arg)
+    elif mode in ["reflect", "symmetric"]:
+        extra_arg = extra_arg or "even"
+        return np.pad(array, pad_width, mode, reflect_type=extra_arg)
+
+
+@wraps(np.pad)
+def pad(array, pad_width, mode, **kwargs):
+    array = asarray(array)
+
+    if mode not in ["constant", "edge", "linear_ramp"]:
+        raise NotImplementError("`pad` does not support the given `mode`.")
```

Comment:
There are various statistical paddings like `maximum`, `mean`, and `minimum`. These could be handled by computing on the global array (possibly sliced if `stat_length` is provided) and then treating them like the `constant` padding case. The `median` case gets a little tricky. Though should be doable as this only gets computed along 1-D pieces of the array.

Things like `reflect`, `symmetric`, and `wrap` can be constructed by slicing the Dask Array into pieces, moving them around, and running `block` to stitch them all together into one Dask Array. Given how this behavior differs from the `pad` behavior included, we may opt to have this be a separate internal function that `pad` calls.

Finally there is the possibility of a user supplying an arbitrary function. We probably could implement this. Have some thoughts on it, but it is enough of a special case that we might want to wait and see if people ask for this feature and see what they expect from it.